### PR TITLE
Issue #2234 remove flags for metrics, logging and service catalog

### PR DESCRIPTION
**Pull request by gbraad**
See https://github.com/openshift/origin/pull/19209 for more info
See #2248 for more info

/cc: @deads2k @jcantrill

### CODE REVIEW

**Comment by praveenkumar** [comment]
@gbraad Since now your removed the flags so might be need to remove those things from the integration feature file also, currently CI is failing for same.

File: cmd/minishift/cmd/start.go [hunk 1]
```diff
@@ -605,9 +605,6 @@ func initClusterUpFlags() *flag.FlagSet {
 	clusterUpFlagSet.AddFlag(cmdUtil.HttpsProxyFlag)
 
 	if minishiftConfig.EnableExperimental {
-		clusterUpFlagSet.Bool(configCmd.Metrics.Name, false, "Install metrics (experimental)")
-		clusterUpFlagSet.Bool(configCmd.Logging.Name, false, "Install logging (experimental)")
-		clusterUpFlagSet.Bool(configCmd.ServiceCatalog.Name, false, "Install service catalog (experimental)")
```
Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: We also need to update the integration test case as well, as of now failing for CI.

Comment (gbraad) [reply cmd/minishift/cmd/start.go]: addressed

File: cmd/minishift/cmd/config/config.go [hunk 2]
```diff
@@ -71,11 +71,8 @@ var (
 	HostPvDir         = createConfigSetting("host-pv-dir", SetString, []setFn{validations.IsValidPath}, nil, true, nil)
 	ServerLogLevel    = createConfigSetting("server-loglevel", SetInt, []setFn{validations.IsPositive}, nil, true, nil)
 	OpenshiftEnv      = createConfigSetting("openshift-env", nil, nil, nil, false, nil)
-	Metrics           = createConfigSetting("metrics", SetBool, nil, nil, true, nil)
-	Logging           = createConfigSetting("logging", SetBool, nil, nil, true, nil)
```
Comment (praveenkumar) [inline cmd/minishift/cmd/config/config.go]: 👍

File: cmd/minishift/cmd/config/config.go [hunk 3]
```diff
@@ -71,11 +71,8 @@ var (
...
 
 	// future enabled flags
-	ServiceCatalog      = createConfigSetting("service-catalog", SetBool, nil, nil, true, nil)
```
Comment (praveenkumar) [inline cmd/minishift/cmd/config/config.go]: 👍

File: test/integration/features/cmd-config.feature [hunk 4]
```diff
@@ -102,19 +102,19 @@ user defined options which changes default behaviour of Minishift.
...
+    | skip-registry-check | t          | true     |
+    | skip-registration   | T          | true     |
+    | skip-registry-check | 1          | true     |
+    | skip-registration   | false      | false    |
+    | skip-registry-check | False      | false    |
+    | skip-registration   | FALSE      | false    |
+    | skip-registry-check | f          | false    |
+    | skip-registration   | F          | false    |
+    | skip-registry-check | 0          | false    |
+    | skip-registration   | f          | false    |
```
Comment (praveenkumar) [inline test/integration/features/cmd-config.feature]: 👍

File: test/integration/features/cmd-config.feature [hunk 5]
```diff
@@ -27,9 +27,9 @@ user defined options which changes default behaviour of Minishift.
 
   Examples: Wrong values for boolean keys
     | property            | value      |
-    | logging             | TRuE       |
-    | metrics             | positive   |
-    | metrics             | yes        |
+    | skip-registration   | TRuE       |
```
Comment (budhrg) [inline test/integration/features/cmd-config.feature]: :+1:

**Comment by LalatenduMohanty** [comment]
@minishift/minishift-dev AFAIK Users need these features need these from Minishift. So IMO we need to provide them in future. May be after the `oc cluster up` restructure is done. I am not sure if we will provide add-ons or flags with `minishift start` at this point.  Just wanted to make sure we agree to this before merging this.

**Comment by gbraad** [comment]
Since they are experimental, it is possible for us to remove a feature when
we see fit.
At the moment we have to many issues resulting from this, and we have no
support story.

**Comment by LalatenduMohanty** [comment]
> I doubt if WE can ever fix this. it would better to solve this as an addon and pass the responsibility to the actual implementers.

I do not think we have any option here. But we will get the input from concerned people.

**Pull request merged**